src/osm_configurator/model/project/calculation/attractivity_phase.py

```diff
@@ -25,4 +55,108 @@ def calculate(self, configuration_manager):
+            result: calculation_state_enum
+            msg: str
+            result, msg = self._calculate_attractivity_in_traffic_cell(cell_name, configuration_manager)
+            if result != calculation_state_enum.CalculationState.RUNNING:
+                return result, msg
+
+        return calculation_state_enum.CalculationState.RUNNING, "running"
+
+    def _calculate_attractivity_in_traffic_cell(self, cell_name: str, config_manager: ConfigurationManager) \
+            -> Tuple[CalculationState, str]:
+        # get the necessary path's
+        reduction_folder: Path = calculation_phase_utility\
+            .get_checkpoints_folder_path_from_phase(config_manager,
+                                                    calculation_phase_enum.CalculationPhase.REDUCTION_PHASE)
+        attractivity_folder: Path = calculation_phase_utility \
+            .get_checkpoints_folder_path_from_phase(config_manager,
+                                                    calculation_phase_enum.CalculationPhase.ATTRACTIVITY_PHASE)
+        input_path: Path = Path(os.path.join(reduction_folder, cell_name + ".csv"))
+        output_path: Path = Path(os.path.join(attractivity_folder, cell_name + ".csv"))
+        # TODO: magic strings
+
+        # Read and create data frames
+        input_df: DataFrame = pd.read_csv(input_path)
+        output_df: DataFrame = self._init_dataframe(config_manager.get_category_manager())
+
+        # Iterate over all elements in this cell
+        index: int
+        element: Series
+        for index, element in input_df.iterrows():
+            try:
+                output_df = self._calculate_attractivity_for_element(element, output_df,
+                                                         config_manager.get_category_manager().get_categories())
+            except CategoryException as err:
+                return calculation_state_enum.CalculationState.ERROR_INVALID_CATEGORIES, err.args[0]
+
+        # Save results as csv
+        output_df.to_csv(output_path)
+
+        return calculation_state_enum.CalculationState.RUNNING, "running"
+
+    def _calculate_attractivity_for_element(self, element: Series, output_df: DataFrame,
+                                            category_list: List[Category]) -> DataFrame:
+        category_name: str = element["category"]
+        category: Category = self._get_category_by_name(category_name, category_list)
+
+        new_entry = pds.Series()
+
+        attractivity: AttractivityAttribute
+        for attractivity in category.get_attractivity_attributes():
+            value: float = attractivity.get_base_factor()
+            attribute: Attribute
+            for attribute in attribute_enum.Attribute:
+                value += attractivity.get_attribute_factor(attribute) * element[attribute.get_name()]
+            new_entry[attractivity.get_attractivity_attribute_name()] = value
+
+        output_df = output_df.append(new_entry, ignore_index=True)
```

Comment:
Dont use append, its very slow see stackoverflow link below(third answer for how slow it is, second answer on how to do it better).

Instead append to a dictinary and then create fro mthe dictionary a dataframe.

Comparsion between append and from dict:
https://stackoverflow.com/questions/10715965/create-a-pandas-dataframe-by-appending-one-row-at-a-time

![Screenshot from 2023-02-07 19-56-08](https://user-images.githubusercontent.com/36456825/217339557-96be3b7c-7e9d-4b93-85b3-4ff996efde31.png)